### PR TITLE
fix(oss-challenge): add persistent volume claim

### PR DESCRIPTION
**Pull request by trowik**
the volume is need for uploading images for the oss-challenge

### CODE REVIEW

File: charts/osschallenge/README.md [hunk 1]
```diff
@@ -56,6 +56,8 @@ This chart is maintained by [Adfinis](https://adfinis.com/?pk_campaign=github&pk
 | osschallenge.settings.emailPort | int | `587` | Port for the SMTP server |
 | osschallenge.settings.emailUser | string | `"example@example.com"` | Username for the SMTP server |
 | osschallenge.settings.siteUrl | string | `"https://oss-challenge.ch"` | URL for OSS-Challenge |
+| persistence.mountPath | string | `"/data"` | specifies where to mount the PV |
+| persistence.size | string | `"10Gi"` | specifies which size the PVC should request |
```
Comment (eyenx) [inline charts/osschallenge/README.md:60]: I'm missing the documentation of `persistence.storageClassName` Have you used `helm-docs` for this?

File: charts/osschallenge/templates/pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: {{ include "osschallenge.fullname" . }}
+  labels:
+    {{- include "osschallenge.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.persistence.storageClassName }}
+  storageClassName: {{ .Values.persistence.storageClassName }}
+  {{- end }}
+  accessModes:
+  - ReadWriteOnce
```
Comment (eyenx) [inline charts/osschallenge/templates/pvc.yaml:12]: the `replicaCount` would be obsolete at this point.

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/osschallenge/README.md (hunk 1 above)
Comment (trowik) [reply charts/osschallenge/README.md:60]: I used the pre-commit, which does use helm-docs

File: charts/osschallenge/templates/pvc.yaml (hunk 2 above)
Comment (trowik) [reply charts/osschallenge/templates/pvc.yaml:12]: So you mean, I should set replicaCount to 0 then? And why would it be obsolete?

File: charts/osschallenge/README.md (hunk 1 above)
Comment (eyenx) [reply charts/osschallenge/README.md:60]: can you recheck? I'm missing this: https://github.com/adfinis-sygroup/helm-charts/pull/174/files#diff-e7cb1aadd92c6e5a2393f8dddc7ceb280e160219d531b142cb850eaff7bfd7b8R113

File: charts/osschallenge/templates/pvc.yaml (hunk 2 above)
Comment (eyenx) [reply charts/osschallenge/templates/pvc.yaml:12]: Not to 0, but having `ReadWriteOnce` hard coded here, would not make it possible to have more than one replica running on different nodes. And if replica aren't running on different nodes, there's really no need for it if the use case would be HA.

Comment (eyenx) [reply charts/osschallenge/templates/pvc.yaml:12]: > The access modes are:
    ReadWriteOnce -- the volume can be mounted as read-write by a single node
    ReadOnlyMany -- the volume can be mounted read-only by many nodes
    ReadWriteMany -- the volume can be mounted as read-write by many nodes

Comment (eyenx) [reply charts/osschallenge/templates/pvc.yaml:12]: Of course the storageProvider needs to support `RWX`. 

https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

File: charts/osschallenge/README.md (hunk 1 above)
Comment (hairmare) [reply charts/osschallenge/README.md:60]: it's commented so it wont get doxed by helm-docs

File: charts/osschallenge/templates/pvc.yaml (hunk 2 above)
Comment (hairmare) [reply charts/osschallenge/templates/pvc.yaml:12]: something like this:
```suggestion
  {{ .Values.persistance.accessModes | toYaml | nindent 2 }}
```

with values:
```yaml
persistance:
  accessModes:
  - ReadWriteOnce
```

Comment (eyenx) [reply charts/osschallenge/templates/pvc.yaml:12]: persistence !

Comment (hairmare) [reply charts/osschallenge/templates/pvc.yaml:12]: oops

File: charts/osschallenge/README.md (hunk 1 above)
Comment (trowik) [reply charts/osschallenge/README.md:61]: Would it be better to uncomment it, set it to nil and make a default-values.yml for the CI?

Comment (eyenx) [reply charts/osschallenge/README.md:61]: Yes.

`storageClassName: ""`

Comment (trowik) [reply charts/osschallenge/README.md:61]: Ok, thanks. Did it like that.

**Comment by trowik** [comment]
I'm not sure why the CI fails. Is it because of line 12?

File: charts/osschallenge/values.yaml [hunk 3]
```diff
@@ -104,3 +104,13 @@ postgresql:
...
   # postgresql.postgresqlPassword -- Set a password for PostgreSQL
   # postgresqlPassword: ~
+
+persistence:
+  # persistence.mountPath -- specifies where to mount the PV
+  mountPath: "/data"
+  # persistence.size -- specifies which size the PVC should request
+  size: "10Gi"
+  # persistence.accessMode -- specifes the access mode for the PV
+  accessMode: ReadWriteOnce
```
Comment (hairmare) [inline charts/osschallenge/values.yaml:114]:
```suggestion
  accessMode:
  - ReadWriteOnce
```

File: charts/osschallenge/templates/pvc.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: {{ include "osschallenge.fullname" . }}
+  labels:
+    {{- include "osschallenge.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.persistence.storageClassName }}
+  storageClassName: {{ .Values.persistence.storageClassName }}
+  {{- end }}
+  accessModes:
+  {{ .Values.persistence.accessMode | toYaml | nindent 2 }}
```
Comment (hairmare) [inline charts/osschallenge/templates/pvc.yaml:12]:
```suggestion
  {{ .Values.persistence.accessModes | toYaml | nindent 2 }}
```

**Review by hairmare — COMMENTED** [review]
The CI environment does not have a storage engine that can fullfill a PVC so wou'll need to disable this part in `ci/` like you did with the postgresql volume in the rmd chart.

File: charts/osschallenge/ci/default-values.yaml [hunk 5]
```diff
@@ -5,3 +5,6 @@ postgresql:
 osschallenge:
   settings:
     emailPassword: s3cr3t
+persistence:
+  storageClassName: "default"
+  enabled: false
```
Comment (trowik) [inline charts/osschallenge/ci/default-values.yaml:10]: This should disable it, right?

Comment (hairmare) [reply charts/osschallenge/ci/default-values.yaml:10]: You need to implement the flag in the `pvc.yaml` template as well:
```
{{- if .Values.persistence.enabled }}
apiVersion: v1
# ...
{{- end }}
```

Comment (trowik) [reply charts/osschallenge/ci/default-values.yaml:10]: ah ok, I see

Comment (hairmare) [reply charts/osschallenge/ci/default-values.yaml:10]: You need to wrap the actual volume stuff in the deployment in a similar `if` block. I think the postgresql chart has an example where an empty volume is used if persistence is disabled.

**Review by hairmare — APPROVED** [review]

**Review by eyenx — APPROVED** [review]

**Pull request merged**